mysql-test/suite/binlog/t/binlog_recover_async_rollback_trx.test

```diff
@@ -0,0 +1,93 @@
+################################################################################
+# Async rollback prepared transactions during binlog crash recovery
+#
+# It verifies that binlog recovery just set the prepared transactions to
+# active and the background recovery rollback thread will rollback the
+# transactions asynchronously.
+################################################################################
+--source include/have_debug.inc
+--source include/have_debug_sync.inc
+--source include/have_innodb.inc
+--source include/have_binlog_format_row.inc
+
+CREATE TABLE t1 (id int primary key, data int) ENGINE = InnoDB;
+INSERT INTO t1 VALUES (0, 1);
+
+--echo #
+--echo # 1. Check DML in prepared state can rollback correctly.
+--echo #
+
+--echo [ connection con1 ]
+--connect(con1, localhost, root,,)
```

Comment:
An `echo` before any `connect` or `connection` is redundant and should be removed. The connections are automatically identified in the `.result` files.